shared/src/types.rs

```diff
@@ -259,11 +259,11 @@ impl<'a> CidrTree<'a> {
         }
     }
 
-    pub fn children(&self) -> impl Iterator<Item = CidrTree> {
+    pub fn children(&self) -> impl Iterator<Item = CidrTree<'_>> {
         self.cidrs
             .iter()
             .filter(move |c| c.parent == Some(self.contents.id))
-            .map(move |c| Self {
+            .map(move |c| CidrTree {
```

Comment:
The reason for this change is that `Self` (which is just a sugar for `CidrTree<'a>`) brings  the the lifetime `'a` to the scope of the returned opaque type. Use `CidrTree` instead to allow rustc to infer a shorter lifetime (no necessarily `'a`).

Generally, it's recommended to use the struct name instead of `Self` to allow rust to infer the appropriate lifetime rather than unnecessarily forcing a specific lifetime on the created

Reply:
Oh this is an interesting one! I've often used `Self` in places like this, good to know the kind of issues that can cause.